tests/test_burst.py

```diff
@@ -0,0 +1,91 @@
+import tempfile
+import xml.etree.ElementTree as ET
+from pathlib import Path
+
+import numpy as np
+import pytest
+import requests
+from shapely import geometry
+
+from hyp3_isce2 import burst
+
+
+URL_BASE = 'https://datapool.asf.alaska.edu/SLC'
+REF_DESC = burst.BurstParams('S1A_IW_SLC__1SDV_20200604T022251_20200604T022318_032861_03CE65_7C85', 'IW2', 'VV', 3)
+SEC_DESC = burst.BurstParams('S1A_IW_SLC__1SDV_20200616T022252_20200616T022319_033036_03D3A3_5D11', 'IW2', 'VV', 3)
+REF_ASC = burst.BurstParams('S1A_IW_SLC__1SDV_20211229T231926_20211229T231953_041230_04E66A_3DBE', 'IW1', 'VV', 4)
+SEC_ASC = burst.BurstParams('S1A_IW_SLC__1SDV_20220110T231926_20220110T231953_041405_04EC57_103E', 'IW1', 'VV', 4)
+
+
+def load_metadata(metadata):
+    metadata_path = Path(__file__).parent.absolute() / 'data' / metadata
+    xml = ET.parse(metadata_path).getroot()
+    return xml
+
+
+@pytest.fixture()
+def tempdir():
+    with tempfile.TemporaryDirectory() as tempdir:
+        yield tempdir
+
+
+def test_create_gcp_df():
+    ref_burst = burst.BurstMetadata(load_metadata('reference_descending.xml'), REF_DESC)
+    n_bursts = int(ref_burst.annotation.findall('.//burstList')[0].attrib['count'])
+    lines_per_burst = int(ref_burst.annotation.findtext('.//{*}linesPerBurst'))
+
+    gcp_df = ref_burst.create_gcp_df()
+    assert np.all(gcp_df.columns == ['line', 'pixel', 'latitude', 'longitude', 'height'])
+    assert gcp_df.line.min() == 0
+    assert gcp_df.line.max() == (n_bursts * lines_per_burst) - 1
+
+
+def test_create_geometry():
+    ref_burst = burst.BurstMetadata(load_metadata('reference_descending.xml'), REF_DESC)
+    burst_number = 3
+    real_box = (100.509898817751, 37.69349213923167, 101.5989880944895, 38.00276647361588)
+
+    gcp_df = ref_burst.create_gcp_df()
```

Comment:
In general I would include the literal value for each input to a function that's being tested, rather than call another function to compute the value. But I don't know what the literal value looks like in this case or how cumbersome it would be to include it, so I'll defer to your judgement.